packages/protractor/src/protractor_web_test.bzl

```diff
@@ -36,37 +37,52 @@ def _protractor_web_test_impl(ctx):
         sibling = ctx.outputs.script,
     )
 
-    files = depset(ctx.files.srcs)
-    for d in ctx.attr.deps:
-        if hasattr(d, "node_sources"):
-            files = depset(transitive = [files, d.node_sources])
-        elif hasattr(d, "files"):
-            files = depset(transitive = [files, d.files])
+    files_depsets = [depset(ctx.files.srcs)]
+    for dep in ctx.attr.deps:
```

Comment:
does this want to be a helper in `providers.bzl`? seems like I've seen the same impl many times?